docs/source/format/CanonicalExtensions.rst

```diff
@@ -72,4 +72,65 @@ same rules as laid out above, and provide backwards compatibility guarantees.
 Official List
 =============
 
-No canonical extension types have been standardized yet.
+Fixed shape tensor
+==================
+
+* Extension name: `arrow.fixed_shape_tensor`.
+
+* The storage type of the extension: ``FixedSizeList`` where:
+
+  * **value_type** is the data type of individual tensors and
+    is an instance of ``pyarrow.DataType`` or ``pyarrow.Field``.
+  * **list_size** is the product of all the elements in tensor shape.
+
+* Extension type parameters:
+
+  * **value_type** = Arrow DataType of the tensor elements
+  * **shape** = shape of the contained tensors as an array
+
+  Optional parameters:
+
+  * **dim_names** = explicit names to tensor dimensions
+    as an array. The length of it should be equal to the shape
+    length and equal to the number of dimensions.
+
+    ``dim_names`` can be used if the dimensions have well-known
+    names and they map to the physical layout (row-major).
+
+  * **permutation**  = indices of the desired ordering of the
+    original dimensions, defined as an array.
+
+    The indices contain a permutation of the values [0, 1, .., N-1] where
+    N is the number of dimensions. The permutation indicates which
+    dimension of the logical layout corresponds to which dimension of the
+    physical tensor (the i-th dimension of the logical view corresponds
+    to the dimension with number ``permutations[i]`` of the physical tensor).
+
+    Permutation can be useful in case the logical order of
+    the tensor is a permutation of the physical order (row-major).
```

Comment:
Shape, permutation and shape all apply to a single element of an array.

Reply:
If you were to have an un-chunked tensor array, then the underlying buffer would have to have the first stride be the largest.

Reply:
I don't think so. The underlying `fixed_size_list` type is just a buffer we by default interpret as row-major. The producer has the option to store data with their own layout and provide the permutation needed to interpret their layout with at read time.

Reply:
Indeed, we still follow the underlying storage type, so operations like slicing, filtering, take etc all work correctly (because indeed otherwise it could not be done as an extension type, but would have to be a new core type). And that's also the reason that the tensor has to be row-major instead of column-major (because otherwise it would violate the integrity of individual elements of the array).